### ISSUE TITLE
yq 3.3.1 comment preservation results in invalid yaml

### ISSUE THREAD

**Issue by samrees** [mikefarah/yq#497]
```
bash-5.0# cat values.yaml
global:
  image:
    tag: # must be overridden
    pullPolicy: IfNotPresent
bash-5.0# /tmp/yq330 write values.yaml global.image.tag 1.1.1 > output.yaml
bash-5.0# cat output.yaml
global:
  image:
    tag: 1.1.1 # must be overridden
    pullPolicy: IfNotPresent
bash-5.0# /tmp/yq331 write values.yaml global.image.tag 1.1.1 > output.yaml
bash-5.0# cat output.yaml
global:
  image:
    tag: # must be overridden
      1.1.1
      pullPolicy: IfNotPresent
```

version of yq: 3.3.1
operating system: amd64

**Comment by samrees** [comment]
To be clear 3.3.0 works fine.

**Comment by mikefarah** [comment]
This is a result of upgrading the underlying go-yaml library. They have made a number of fixes around handling comments, and this looks like it's a side-effect of those changes.

From debugging that library, I can see that the comment is being associated as a line comment of the _key_ 'tag' rather than it's value (as there is no value I guess).

When we specific a value for that entry, it goes on a new line to accommodate the comment for the key node :(

I'll raise a bug against go-yaml and see what happens

**Comment by mikefarah** [comment]
https://github.com/go-yaml/yaml/issues/656

**Comment by ogruene** [comment]
For merging two yamls we currently have the same effect. 
```
root:
  sub1: #comment
  sub2: #comment
```
and
```
root:
  sub1: value1
  sub2: value2
```
Using `yq m -x values-1.yaml values-2.yaml` with yq 3.4.0 results in
```
root:             
  sub1: #comment  
    value1        
    sub2: #comment
      value2
``` 
And with yq 3.3.0 in
```
root:
  sub1: value1 #comment
  sub2: value2 #comment
```

Oliver Grüneberg <<oliver.grueneberg@daimler.com>>, Daimler TSS GmbH
[Legal Imprint](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)

**Comment by mikefarah** [comment]
Fixed in 4.3.0
